ex4B/ex4.js

```diff
@@ -0,0 +1,104 @@
+/**
+ * Write a function that convers a string into number; return null if the string
+ * doesn't contain a valid number.
+ *
+ * For example:
+ * toNumber("71") === 71
+ * toNumber("a7") === null
+ */
+function toNumber(str) {
+
+	var strNum = '', strDec = '', number = null, i = 0, seenDot = false, negative = false;
+	const CODE_FOR_0 = "0".charCodeAt();
+	// Negative sign can appear only at the start of string and followed by number
+	//if (str.slice(0, 1) === '-' && isNumber(str.slice(1, 2))) {
+	
+	if (str.charAt(0) === '-' ) {
+		
+		if (isNumber(str.charAt(1))) {
+
+			negative = true;  
+			i = 1;
+
+		} else {
+		
```

Comment:
Even though we discussed it, I'd put a comment here saying: `// Will catch "-.3", for example.